components/EventsTable.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import { StyleSheet, Text, View } from "react-native";
+import { DataTable } from "react-native-paper";
+
+import { useQuery, gql } from "@apollo/client";
+import { FETCH_EVENTS_QUERY } from ".././util/graphql";
+
+function EventsTable() {
+  let { data, loading } = useQuery(FETCH_EVENTS_QUERY);
+  let events = null;
+  let tableContents = [];
+
+  if (data) {
+    events = data.getEvents;
+
+    const maxEvents = Math.min(events.length, 5);
+
+    const monthOptions = require("./../assets/options/month.json");
+    const month = new Date().getMonth();
+    const semester = monthOptions[month].value;
+
+    for (let i = 0; i < maxEvents; i++) {
+      const event = events[i];
+      if (event.semester == semester) {
+      const date = event.expiration.substring(0, 15);
+        let row = [
+          <DataTable.Row>
```

Comment:
Everything is fine, but I'm getting a warning saying that each child of tableContents should have a key, look at this link for more details: [https://reactjs.org/docs/lists-and-keys.html#keys](url)